### DISCUSSION TITLE
DDC control not working with M2 Pro/Max systems

### DISCUSSION THREAD

**Discussion by ashishbl86** [MonitorControl/MonitorControl#1299] [General]
I have 2 external monitors LG 27UL850 and LG 27UL650. I was previously using them with MacBook Pro 2017 (Intel Core i5) and monitor control was working fine.

I just got the new MacBook Pro 14 with M2 Pro chip and monitor control is not working in DDC mode. Both the monitors are connected directly to the laptop. When using the external monitors I keep the Macbook's lid closed. And the monitors are used as main monitor and extended monitor config.

To debug I tried to use m1ddc and it's working but not completely.

E.g. m1ddc display list didn't give any output. But when I tried m1ddc get luminance it gave me correct brightness of my default monitor. And when I ran set luminance the brightness of my default monitor changed successfully. There was no change on my second monitor. The same observation was for volume commands as well.

**Comment by waydabber** [comment]
Hi there, can you try this one and give me some feedback if it works? If you can, please try both USB-C and HDMI!

https://github.com/waydabber/BetterDisplay/releases

Thank you!

(note: things has changed for the M2 Pro/Max, so MC and m1ddc is incompatible as well as Arm64DDC. Just finished updating BD which hopefully works - not that I have an M2 Pro/Max Mac - if things prove to be stable and working properly, I'll eventually update all opensource stuff as well!)

**Reply by ashishbl86** [reply]
Thank you so much. I tried BD and it works absolutely fine. I will try to find an HDMI cable and test that too. Right now both are connected with USB-C.

One monitor (LG 27UL850) has a USB-C port and for the other monitor (LG 27UL650) doesn't has a USB-C is connected via USB-C to DisplayPort adapter.

**Reply by waydabber** [reply]
Thanks @ashishbl86 for the feedback!

**Reply by ashishbl86** [reply]
Hey @waydabber. I tested with HDMI cable and it too works. I didn't do extensive testing but it works for my basic requirement of brightness and volume controls. Thanks a ton again for making BetterDisplay.

**Comment by jflynn1000** [comment]
I found this thread as a former MonitorControl user on an intel MacBook Pro who also just updated to an M2 Pro MacBook. @waydabber, your suggestion works perfectly for me as well! Here's my setup: 14 inch MacBook Pro with M2 Pro, thunderbolt out to a Belkin dock, though a G-Raid external hard drive, and into an LG 40WP95C monitor. I installed BetterDisplay this morning before work, and its been flawless all day. Thanks for the help!

**Comment by lyncasterc** [comment]
Same problem with a M2 Max Macbook Pro and a Dell U2415, connected via Thunderbolt through a Kensington dock. Lunar and MonitorControl worked just fine on my Intel Macs but they stopped working when I got this one. Installed BetterDisplay and it seems to be working well so far.

**Comment by NoobCanoe** [comment]
Same issue. Previous setup, Intel MacBook Pro connected to a BenQ EW3280U via USB-C where MonitorControl worked flawlessly. Replaced the MacBook. Pro with an M2 Pro Mac Mini and lost the ability to control volume. I installed BetterDisplay 1.3.14, and I can again control the volume. Will MonitorControl also be updated?

**Reply by waydabber** [reply]
Yes, updating all open source stuff I am part of with M2 Pro compatibility is the plan. Want to do it methodically though - here is the approach I am aiming for: https://github.com/waydabber/BetterDisplay/issues/1363#issuecomment-1400107981 . Will eventually get there, but I am in the middle of some other stuff right now. Of course any contributor is free to work on making this happen meanwhile!

Note: although some advanced features of BetterDisplay (like XDR extra brightness, native smooth scaling etc.) are Pro only, all DDC and brightness control stuff is free so you can keep using it without worry (or any nag screen as there is no any in BD) until MC is updated. :)

**Reply by NoobCanoe** [reply]
Thank you!

**Reply by wklhtcone** [reply]
Yes, I have the same problem with Mac mini(M2 Pro) and LG 27UK850, but the M2 version Mac mini doesn't have this issue

**Comment by pharwood** [comment]
Same issue. Upgraded from Mac mini M1 to Mac mini M2 Pro with LG 27UP850-W. MonitorControl worked great with the M1, but fails to detect the monitor as Hardware (DDC) on the M2. Will try BetterDisplay as suggested.

**Reply by pharwood** [reply]
BetterDisplay works great.

**Comment by danwetherald** [comment]
I can also confirm only brightness can be controlled on my new M2 Macbook Pro.

Following.

**Comment by chaparrodaniel** [comment]
I experienced the same issue with MonitorControl volume control not supporting my LG 40WP95C-W 40 display.

I can confirm that BetterDisplay supports volume control on my MBP 2023 with M2 Max silicon and an LG 40WP95C-W 40 display.

@waydabber 
Im a little confused on the development forks between MonitorControl and BetterDisplay.

- **Is MonitorControl going to support volume control on a LG 40WP95C-W 40?**
- **Are you part of dev team working on MonitorControl?**

I really like the simplicity of MonitorControl. BetterDisplay has so many settings, that I was lost trying to locate the volume settings.

**Reply by waydabber** [reply]
Both MC and BD should support volume control on the LG 40WP95C-W 40 _if_ the display is compatible with DDC volume control. This discussion is about general DDC support for M2 Pro/Max system (also see the relevant issue: https://github.com/MonitorControl/MonitorControl/issues/1301)

To question: I am one of the maintainers of MonitorControl (+ did a lot of development in it, but nowadays I work mostly on BetterDisplay). MonitorControl is developed by the open source community so anybody can improve the app with M2 Pro/Max support who wishes to put some effort into it. If this does not happen, then it will eventually be my privilege to do so. :)

_(note: if you have any questions about how to change BetterDisplay settings, the best solution is to hop over to that repo's discussion and ask your questions or join the app's discord channel.)_

**Reply by chaparrodaniel** [reply]
Thanks for the quick response [@waydabber](https://github.com/waydabber)

I can confirm that MonitorControl 4.10 does not support volume control on my MBP 2023 with M2 Max silicon and an LG 40WP95C-W 40 display.

OS: MacOS Ventura 13.0
MBP 2023 14 inch with M2 Max chip
RAM: 32GB
Monitor: LG 40WP95C-W 40

If this does not qualify my post in the DDC volume control discussion, please let me know of the correct discussion thread to post my feedback.

Thanks again for the quick response.

**Reply by waydabber** [reply]
It's ok, the issue is with the M2 Max system, not the display (the fact that it's about the M2 was not obvious from the original edition of the post, I thought it's a generic display issue). MC does not currently support DDC control on the M2 Max.

**Reply by chaparrodaniel** [reply]
Thanks for the explanation @waydabber

**Comment by wklhtcone** [comment]
Yes, I have the same problem with Mac mini(M2 Pro) and LG 27UK850, but the M2 version Mac mini doesn't have this issue

**Comment by ariromano** [comment]
same problem here with M2 max MacBook pro and 3 Gigabyte M32UC Displays :/ two are connected via USB-C → DisplayPort and one via USB-C → HDMI

**Reply by waydabber** [reply]
You can use this (DDC stuff is free) until support is added to MonitorControl: https://github.com/waydabber/BetterDisplay

**Comment by TinkoLiu** [comment]
I just made a [simple commandline tool](https://github.com/TinkoLiu/DDCCli) by using the code from #1404 and works well with my monitor on M2 Pro mac mini.
